examples/thermostat/linux/thermostat-schedules-manager.cpp

```diff
@@ -0,0 +1,105 @@
+#include <app/clusters/thermostat-server/thermostat-server.h>
+
+#include <vector>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters::Thermostat;
+
+static std::vector<PresetStruct::Type> gsActivePresets;
+static std::vector<PresetStruct::Type> gsEditingPresets;
+
+static void 
+onEditStart(ThermostatMatterScheduleManager * mgr, ThermostatMatterScheduleManager::editType type)
+{
+    ChipLogProgress(Zcl, "ThermstatScheduleManager - onEditStart %s", type == ThermostatMatterScheduleManager::Presets ? "Presets" : "Schedules");
+}
+
+static void 
+onEditCancel(ThermostatMatterScheduleManager * mgr, ThermostatMatterScheduleManager::editType type)
+{
+    ChipLogProgress(Zcl, "ThermstatScheduleManager - onEditCancel %s", type == ThermostatMatterScheduleManager::Presets ? "Presets" : "Schedules");
+}
+
+static void 
+onEditCommit(ThermostatMatterScheduleManager * mgr, ThermostatMatterScheduleManager::editType type)
+{
+    ChipLogProgress(Zcl, "ThermstatScheduleManager - onEditCommit %s", type == ThermostatMatterScheduleManager::Presets ? "Presets" : "Schedules");
+    gsActivePresets = gsEditingPresets;
```

Comment:
This looks pretty broken; how does this support editing presets _and_ schedules at the same time?

Also, why is cancel a no-op, etc?

Please make sure issues are filed on all the bugs here.

Reply:
sorry, this all WIP.  Maybe shouldn't have created PR, was more checking to see what breaks with autobuilds.